plugins/replay/src/main/java/forklift/replay/ReplayESWriter.java

```diff
@@ -1,94 +1,167 @@
 
     public ReplayESWriter(String hostname) {
-        this(hostname, 9300, "elasticsearch");
+        this(hostname, 9200, "elasticsearch");
     }
 
+    // clusterName is retained in constructor for compatibility
     public ReplayESWriter(String hostname, int port, String clusterName) {
-        final Settings settings = Settings.settingsBuilder()
-                        .put("cluster.name", clusterName).build();
-
-        this.client = TransportClient.builder()
-            .settings(settings)
-            .build()
-            .addTransportAddress(new InetSocketTransportAddress(new InetSocketAddress(hostname, port)));
+        this.restClient = RestClient.builder(new HttpHost(hostname, port, "http"))
+            .setRequestConfigCallback(requestConfig -> requestConfig
+                                                           .setConnectTimeout(3_000)
+                                                           .setSocketTimeout(20_000))
+            .setDefaultHeaders(new Header[] {
+                 new BasicHeader("Accept", "application/json; charset=utf-8"),
+                 new BasicHeader("Content-Type", "application/json; charset=utf-8")
+             })
+            .setMaxRetryTimeoutMillis(20_000)
+            .build();
     }
 
     @Override
-    protected void poll(ReplayESWriterMsg t) {
-        final String replayVersion = t.getFields().get("forklift-replay-version");
+    protected void poll(ReplayESWriterMsg replayMessage) {
+        final String replayVersion = replayMessage.getFields().get("forklift-replay-version");
         if ("3".equals(replayVersion)) { // latest version
-            String indexDate = t.getFields().get("first-processed-date");
-            if (indexDate == null)
-                indexDate = LocalDate.now().format(DateTimeFormatter.BASIC_ISO_DATE);
-            final String index = "forklift-replay-" + indexDate;
+            processVersion3Replay(replayMessage);
+        } else if ("2".equals(replayVersion) || replayVersion == null) { // older versions didn't have the replay version or didn't persist it with the message
+            processVersion2OrEarlierReplay(replayMessage);
+        } else {
+            log.error("Unrecognized replay version: '{}' for message ID '{}' with fields '{}'",
+                      replayVersion, replayMessage.getId(), replayMessage.getFields());
+        }
+    }
 
-            try {
-                // Index the new information.
-                client.prepareIndex(index, "log")
-                    .setVersion(t.getVersion()).setVersionType(VersionType.EXTERNAL_GTE)
-                    .setId(t.getId()).setSource(t.getFields()).execute().actionGet();
-            } catch (VersionConflictEngineException expected) {
-                log.debug("Newer replay message already exists", expected);
+    private void processVersion3Replay(final ReplayESWriterMsg replayMessage) {
+        String indexDate = replayMessage.getFields().get("first-processed-date");
```

Comment:
Maybe we should start a class for forklift managed metadata so we can avoid using strings? Probably doesn't matter in this pr but something to think about.

Reply:
+1, that seems like a good idea